osu.Desktop/DiscordRichPresence.cs

```diff
@@ -96,6 +98,21 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private string truncate(ReadOnlySpan<char> str)
+        {
+            if (Encoding.UTF8.GetByteCount(str) <= 128)
+                return new string(str);
+
+            int ellipsisLength = Encoding.UTF8.GetByteCount(new[] { '…' });
```

Comment:
Should be constant or static.

Reply:
Additional note: I'm not sure about acceptance for the ellipsis character. It's 3 bytes in UTF8 like a common wide character, same with three dots `...`.
It's easy to input when using Chinese IME, and often used in Chinese text. However, in English text, I'm seeing three dots used much more.

Reply:
it's more correct to use this character, but not easily accessible to normal users (no key binding). for automated cases i believe it should be preferred.